### DISCUSSION TITLE
Public OpenAPI YAML specification

### DISCUSSION THREAD

**Discussion by MartinDelille** [speechmatics/community#12] [Ideas & suggestions]
It would be nice to have a repository containing the openapi yaml specification so we can use generator like https://openapi-generator.tech/ for the client code.

**Comment by TudorCRL** [comment]
Hi @MartinDelille thanks for the feedback. You should be able to get a copy of the schema for the batch and realtime transcription APIs from our JS SDK [here](https://github.com/speechmatics/speechmatics-js-sdk/tree/main/schemas). We used these schemas to generate the TS types for our SDK. It's also possible to download the schema from our [docs site](https://docs.speechmatics.com/jobsapi) but that's only for batch. I think we'll be adding the realtime schema to our docs site at some point soon.

**Reply by MartinDelille** [reply]
Thank you very much!

Do you think you could publish it in a separate repository and add it as submodule ? This would avoid to clone the whole JS SDK when you want to generate a client in another language!

Since the JS SDK is very lightweight this is not a big issue, that being said.

**Reply by MartinDelille** [reply]
Hi,

Did you considered my idea versionning this specification separately ? This would help people using C++ generators for example.

**Reply by TudorCRL** [reply]
Hi @MartinDelille we're looking into building out a process for publishing our API specs publicly into their own GitHub repo. I'll make sure to get back to you with any progress
